mozilla/bedrock
# Add pre-commit hook for including the MPLv2 in all source code files

Adds to Python, JS, SCSS, Jinja HTML, Fluent files and shell scripts

## Description
This changeset adds a pre-commit hook that can automatically insert a licence in a file. It is configured to insert the MPLv2 into Python, JS, SCSS, Jinja HTML and shell files.

## Issue / Bugzilla link

Resolves #10614

## Testing

* pull this branch
* Pick a Python, JS, SCSS, .sh, Fluent, Jinja HTML file and remove the MPLv2 text from it
* Stage the file(s)
* `git commit` to trigger the pre-commit hooks
* Confirm that all the files fail pre-commit checks and have the MPLv2 text re added to them in a way that staging the changes removes any difference
* run `pre-commit run --all-files` and confirm there are no changes
* Test drive around the site and confirm there are no unexpected markup bombs or JS errors
